control/jet_vane_playground.cc

```diff
@@ -0,0 +1,195 @@
+
+  constexpr double vane_x = 20e-3;
+  constexpr double vane_z = 30e-3;
+
+  // These are all in the *angled* vane frame
+  const std::vector<jcc::Vec3> vane_points_vane_frame = {{
+      jcc::Vec3(-vane_x * ONE_QUARTER, 0.0, -vane_z),
+      jcc::Vec3(-vane_x * ONE_QUARTER, 0.0, 0.0),
+      jcc::Vec3(0.5 * vane_x, 0.0, 0.0),
+      jcc::Vec3(vane_x * THREE_QUARTERS, 0.0, -0.2 * vane_z),
+      jcc::Vec3(vane_x * THREE_QUARTERS, 0.0, -vane_z),
+  }};
+
+  for (const auto& pt_plane_frame : vane_points_vane_frame) {
+    poly.points.push_back(world_from_plane * (pt_plane_frame + vane_tip_edge_from_cp));
+  }
+  poly.outline = true;
+  poly.color = jcc::Vec4(0.8, 0.8, 0.8, 0.8);
+  geo.add_polygon(poly);
+
+  geo.add_axes({world_from_plane, 0.005, 3.0});
+}
+
+SE3 vane_unit_from_vane(const double angle, const VaneConfiguration& cfg) {
+  const SE3 zero_from_angle = jcc::exp_z(angle);
+  return cfg.T_vane_unit_from_zero * zero_from_angle;
+}
+
+void put_quadraframe(viewer::SimpleGeometry& geo,
+                     const QuadraframeStatus& status,
+                     const QuadraframeConfiguration& quad_cfg,
+                     const VaneConfiguration& vane_cfg) {
+  const SE3 com_from_oriented_vane_0 =
+      quad_cfg.com_from_vane_unit_0 * vane_unit_from_vane(status.servo_0_angle, vane_cfg);
+  put_plane(geo, com_from_oriented_vane_0);
+
+  const SE3 com_from_oriented_vane_1 =
+      quad_cfg.com_from_vane_unit_1 * vane_unit_from_vane(status.servo_1_angle, vane_cfg);
+  put_plane(geo, com_from_oriented_vane_1);
+
+  const SE3 com_from_oriented_vane_2 =
+      quad_cfg.com_from_vane_unit_2 * vane_unit_from_vane(status.servo_2_angle, vane_cfg);
+  put_plane(geo, com_from_oriented_vane_2);
+
+  const SE3 com_from_oriented_vane_3 =
+      quad_cfg.com_from_vane_unit_3 * vane_unit_from_vane(status.servo_3_angle, vane_cfg);
+  put_plane(geo, com_from_oriented_vane_3);
+}
+
+}  // namespace
+
+void go1() {
+  setup();
+  const auto view = viewer::get_window3d("Mr. Vane, vanes");
+  const auto geo = view->add_primitive<viewer::SimpleGeometry>();
+
+  //
+  // Generate a force sample
+  //
+
+  const JetStatus jet_status({.throttle = 1.0});
+  const VaneConfiguration vane_cfg;
+  const JetConfiguration jet_cfg;
+  const QuadraframeConfiguration quad_cfg = {};
+
+  const double max_angle_rad = 0.2268;
+  double change = 0.01;
+  double theta_0_rad = -max_angle_rad;
+  while (!view->should_close()) {
+    theta_0_rad += change;
+    if (std::abs(theta_0_rad) > max_angle_rad) {
+      change = -change;
+      theta_0_rad += change;
+    }
+
+    const QuadraframeStatus quad_status = {theta_0_rad, 0.0, 0.0, 0.0};
+    put_quadraframe(*geo, quad_status, quad_cfg, vane_cfg);
+
+    const Wrench com_wrench = total_wrench_com_frame(jet_status, quad_status, vane_cfg, jet_cfg, quad_cfg);
+
+    constexpr double LINE_WIDTH = 4.0;
+    const jcc::Vec4 force_color(1.0, 0.5, 0.0, 0.8);
+    const jcc::Vec4 torque_color(0.0, 1.0, 1.0, 0.8);
```

Comment:
Yeah you would be happier with a color library.  Lots of graphics frameworks have these, and if you limited yourself to a nice palette it would end up looking great.